clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -440,7 +454,8 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                           TRACE_MARKER_TYPE_KERNEL_XFER)) ||
                     // We expect a gap on a window transition.
                     shard->window_transition_ ||
-                    shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER,
+                    shard->prev_instr_.instr.type == TRACE_TYPE_INSTR_SYSENTER ||
+                    saw_double_syscall_instrs_with_same_pc,
```

Comment:
Add a comment clarifying that we skip this check because we don't want to double-report.